pycaching/cache.py

```diff
@@ -674,14 +684,91 @@ def load_quick(self):
 
         logging.debug("Cache loaded: {}".format(self))
 
+    def load_by_guid(self):
+        """Load cache details using the GUID to request and parse the caches
+        'print-page'. Loading as many properties as possible except the
+        following ones, since they are not present on the 'print-page':
+
+          + original_location
+          + state
+          + found
+          + pm_only
+
+        :raise .PMOnlyException: If the PM only warning is shown on the page
+        """
+        if not hasattr(self, "guid"):
+            self._load_guid()
+        res = self.geocaching._request(self._urls["print_page"],
+                                       params={"guid": self.guid})
+        if res.find("p", {"class": "Warning"}) is not None:
+            raise errors.PMOnlyException()
+        content = res.find(id="Content")
+
+        self.name = content.find("h2").get_text()
+
+        self.location = Point.from_string(
+            content.find("p", {"class": "LatLong Meta"}).text)
+
+        type_img = os.path.basename(content.find("h2").img.get("src"))
+        self.type = Type.from_filename(os.path.splitext(type_img)[0])
+
+        size_img = content.find("img", src=re.compile("\/icons\/container\/"))
+        self.size = Size.from_string(size_img.get("alt").split(": ")[1])
+
+        D_and_T_img = content.find(
+            "p", {"class": "Meta DiffTerr"}).find_all("img")
+        self.difficulty, self.terrain = [
+            float(img.get("alt").split()[0]) for img in D_and_T_img
+        ]
+
+        self.author = content.find(
+            "p", text=re.compile("Placed by:")).text.split("\r\n")[2].strip()
+
+        self.hidden = content.find(
+            "p", text=re.compile("Placed Date:")).text.split(": ")[1]
```

Comment:
Please make this a little bit more solid. This can easily break, if some user sets a date format containing `: `.